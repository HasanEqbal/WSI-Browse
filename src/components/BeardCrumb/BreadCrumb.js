import React from 'react';
import './BreadCrumb.scss';

export default function BreadCrumb() {
  return (
    <React.Fragment>
      <ul className="breadcrumbs">
        <li><a href="#">Home</a></li>
        <li><a href="#">Features</a></li>
        <li className="disabled">Gene Splicing</li>
        <li>
          <span className="show-for-sr">Current: </span> Cloning
    </li>
      </ul>
    </React.Fragment>
  )
}
