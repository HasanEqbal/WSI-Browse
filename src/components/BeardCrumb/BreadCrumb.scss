.breadcrumbs{
  margin: auto !important;
  margin-left: 10%;
}

.grid-x{
  width: 79.8%;
}
