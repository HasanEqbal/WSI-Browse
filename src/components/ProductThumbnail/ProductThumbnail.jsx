import React, { useState } from 'react';
import Parser from 'html-react-parser';
import { Link } from 'react-router-dom';
import './ProductThumbnail.scss';

export default function ProductThumbnail({ product }) {
  const {
    name,
    id,
    price,
    priceRange,
  } = product;

  let regularHigh
  let regularLow
  let sellingHigh
  let sellingLow

  if (priceRange) {
    if (priceRange.regular) {
      regularHigh = priceRange.regular.high;
      regularLow = priceRange.regular.low;
    }

    if (priceRange.selling) {
      sellingHigh = priceRange.selling.high;
      sellingLow = priceRange.selling.low;
    }
  }

  let regularPrice
  let sellingPrice

  if (price) {
    regularPrice = price.regular
    if (price.selling) {
      sellingPrice = price.selling
    }
  }

  let image = product.thumbnail.href;

  return (
    <React.Fragment>
      <div className="cell thumbnail">
        <div className="product-card" >
          <div className="product-card-thumbnail">
            <Link className="product-main-image" to={`/shop/${id}`}><img src={image} alt={name} /></Link>
          </div>
          <h6 className="product-card-title">
            <Link className="product-name"to={`/shop/${id}`}>{Parser(name)}</Link></h6>
          <div>
            {regularHigh ?
              <span className="product-card-old-price">
                Original Price: ${regularLow}-${regularHigh}
              </span>
              : regularPrice ?
                <span className="product-card-regular-price">
                  Original Price: ${regularPrice}
                </span> : ""
            }
          </div>
          <div>
            <span className="product-card-sale-price">
              Sale Price: ${sellingHigh
                ? `${sellingLow} - $${sellingHigh}`
                : sellingPrice
              }
            </span>
          </div>
        </div>
      </div>
    </React.Fragment >
  )
}

