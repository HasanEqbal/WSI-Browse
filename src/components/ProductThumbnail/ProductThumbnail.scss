
.product-card {
  background-color: #fefefe;
  padding: 1rem;
  margin-bottom: 1.5rem;
  margin-right: .8rem;
  border: #D9524F;
}

.product-card-thumbnail {
  display: inline;
  position: relative;
  background-color: #f3f3f3;
}

.product-card-thumbnail:hover{
  border: #D9524F;
}
.product-card-title {
  font-family: 'Open sans', sans-serif;
  color: #082a4c;
  font-size: 0.9rem;
  font-weight: 600;
  margin-top: 1rem;
  height: 4rem;
  margin-bottom: 0;
}


.product-card-sale-price {
  color: #D9524F;
  display: inline-block;
  font-size: 0.75rem;
  font-weight: 600;
  margin-top: 0.8rem;
  margin-left: 0.2rem;
  text-align: center;
}

.product-card-old-price {
  color: darken(#082a4c, 30%);
  display: inline-block;
  font-size: 0.80rem;
  margin-left: 0.3rem;
  margin-right: 0.3rem;
  text-decoration: line-through;
  text-align: center;
}

.product-card-colors {
  display: block;
  margin-top: 0.8rem;
}

.product-card-color-option {
  display: inline-block;
  height: 25px;
  width: 25px;
}

.thumbnail{
  flex-basis: 30%;
}

