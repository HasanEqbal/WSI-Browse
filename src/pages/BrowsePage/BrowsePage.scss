.grid-x{
  width: 80%;
  margin-left: 10%;
}

.loading-indicator {
  margin-left: 30%;
  align-content: center;
}
