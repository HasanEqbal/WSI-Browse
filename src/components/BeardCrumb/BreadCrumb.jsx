import React from 'react';
import './BreadCrumb.scss';

export default function BreadCrumb({ categoryname }) {
  return (
    <React.Fragment>
      <ul className="breadcrumbs grid-x grid-padding-x large-up-3">
        <li><a href="#">Cookware</a></li>
        <li><a href="#">Features</a></li>
        <li className="disabled">Gene Splicing</li>
        <li>
          <span className="show-for-sr">Current: </span> {categoryname}
        </li>
      </ul>
    </React.Fragment>
  )
}
