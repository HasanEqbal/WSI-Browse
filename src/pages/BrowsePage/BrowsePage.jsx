import React from 'react';
import BreadCrumb from '../../components/BeardCrumb/BreadCrumb';
import ProductThumbnail from '../../components/ProductThumbnail/ProductThumbnail';
import './BrowsePage.scss';

function BrowsePage({ productsList }) {
  let { groups, name } = productsList;
  return (
    <React.Fragment>
      <BreadCrumb categoryname={name} />
      <div className="grid-x grid-padding-x small-up-1 medium-up-3 large-up-3">
        {groups ? groups.map((item, index) => {
          return (<ProductThumbnail key={index} product={item} />)
        }) : <img src="../../ajax-loader.jpg"></img>}
      </div>
    </React.Fragment>
  );

}

export default BrowsePage;
