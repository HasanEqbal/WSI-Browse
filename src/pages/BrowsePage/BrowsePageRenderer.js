import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getProducts } from '../../actions';
import ProductThumbnail from '../../components/ProductThumbnail/ProductThumbnail';
import './BrowsePage.scss';
import BreadCrumb from '../../components/BeardCrumb/BreadCrumb';

class BrowsePageRenderer extends Component {
  componentDidMount() {
    this.props.getProducts();
  }

  render() {
    return (
      <React.Fragment>
        <BreadCrumb />
        <div className="row product-thumnail-container">
          {this.props.productsList ? this.props.productsList.map((item) => { return (<ProductThumbnail product={item} />) }) : <img src="../../ajax-loader.jpg"></img>}
        </div>
      </React.Fragment>
    );
  }
}

export function mapStateToProps(state) {
  return { productsList: state.productsList.groups };
}

export default {
  component: connect(mapStateToProps, { getProducts })(BrowsePageRenderer)
};
