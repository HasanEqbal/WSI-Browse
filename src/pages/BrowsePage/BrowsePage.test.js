import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import BrowsePage from './BrowsePage';
import BreadCrumb from '../../components/BeardCrumb/BreadCrumb';
import ProductThumbnail from '../../components/ProductThumbnail/ProductThumbnail';
const fixture = require('../../testData/testFixture.json');

describe('Component : BrowsePage', () => {

  it('expect to render BrowsePage component', () => {
    expect(shallow(<BrowsePage productsList="productsList" />)).toMatchSnapshot();
  })

  it('verifying rendering of BreadCrumb component', () => {
    const productlist = { name: "All New" }
    const wrapper = mount(<BrowsePage productsList={productlist} />)
    expect(wrapper.contains(<BreadCrumb categoryname={productlist.name} />)).toBe(true);
  })

  it('verifying rendering of productThumbnail component', () => {
    const groups = fixture.groups
    const wrapper = mount(
      <MemoryRouter>
        <BrowsePage productsList={fixture} />
      </MemoryRouter>
    )
    expect(wrapper.contains(<ProductThumbnail product={groups[0]} />)).toBe(true);
    expect(wrapper.contains(<ProductThumbnail product={groups[groups.length - 1]} />)).toBe(true);
  })

  it('verifying image loader display when data is loading', () => {
    const productlist = {}
    const wrapper = mount(<BrowsePage productsList={productlist} />)
    expect(wrapper.find('img.loading-indicator').length).toBe(1);
  })

})
