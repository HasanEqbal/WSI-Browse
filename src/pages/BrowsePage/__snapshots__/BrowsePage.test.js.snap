// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component : BrowsePage expect to render BrowsePage component 1`] = `ShallowWrapper {}`;
